### PR TITLE
[DRAFT] add casm json target

### PR DESCRIPTION
**Pull request by tohrnii**
Addressing #676

### CODE REVIEW

File: scarb/src/compiler/compilers/lib.rs [hunk 1]
```diff
@@ -127,3 +149,16 @@ impl Compiler for LibCompiler {
         Ok(())
     }
 }
+
+/// Represents a contract in the Starknet network.
+#[derive(Clone, Default, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct SerializedCasm {
```
Comment (maciektr) [inline scarb/src/compiler/compilers/lib.rs:155]: Why don't we just define some serialization for the struct in Cairo repo (in that repo)? https://github.com/starkware-libs/cairo/blob/233c0fdabe8c5010d10b5d16e02fecf6b656ef6b/crates/cairo-lang-sierra-to-casm/src/compiler.rs#L95

**Review by maciektr — COMMENTED** [review]
Hi!

File: scarb/src/compiler/compilers/lib.rs (hunk 1 above)
Comment (tohrnii) [reply scarb/src/compiler/compilers/lib.rs:155]: You're right it should ideally go there but I wasn't sure if the cairo team wanted to make this a part of that repo based on some discussion in related issues. I'll open a PR there.

**Comment by maciektr** [comment]
Now that I think about it, please consider if your usage isn't already covered by a tool called `universal-sierra-compiler` https://github.com/software-mansion/universal-sierra-compiler/tree/master, which is already capable of compiling Sierra json to Casm json format (and has not been available when the issue has been created).

**Comment by tohrnii** [comment]
@maciektr Thank you. That tool should already cover the use case. I'll close this PR.

**Comment by tohrnii** [comment]
@maciektr After discussion with @rodrigo-pino, there are a couple things missing from the generated casm json, namely entrypoint and builtins. It might be better for scarb to produce the json file directly instead of relying on yet another tool since it already generates the sierra json file. What do you think?

**Comment by Arcticae** [comment]
@tohrnii It can be added to the `universal-sierra-compiler` if you need it. Happy to accept any PRs there :)

**Comment by maciektr** [comment]
Hi @tohrnii!

Very sorry for the delay in response, we had a national holidays here in Poland, so I've been out of office for a bit. 

> After discussion with @rodrigo-pino, there are a couple things missing from the generated casm json by the universal-sierra-compiler, namely entrypoint and builtins. 

I believe this should be added to USC then. I second @Arcticae here. 

> It might be better for scarb to produce the json file directly instead of relying on yet another tool since it already generates the sierra json file. 

We do see Scarb as primarily Cairo -> Sierra compiler, and the CASM is just an addition here. We recommend using the USC wherever you rely on CASM programatically and the builtin Scarb feature for by-hand debuging.  We move our tooling to use USC as well. 

Would it be ok for you?

**Comment by tohrnii** [comment]
Thank you @maciektr @Arcticae for the confirmation. Sounds good, I'll open a PR there. Closing this one then.

**Pull request closed without merging**